File: launchschool_exercises/210_small_problems/medium2/unlucky-days.js
/*
  Write a function that takes a year as an argument, and returns the number of 'Friday the 13ths' in that year. You may assume that the year is greater than 1752 (when the modern Gregorian Calendar was adopted by the United Kingdom). You may also assume that the same calendar will remain in use for the foreseeable future.
*/
function fridayThe13ths(year) {
  return [0,1,2,3,4,5,6,7,8,9,10,11].filter(function (month) {
    var date = new Date(year, month, 13);

    if (date.getDay() === 5) {
      return true;
    }
  }).length;
}

console.log(fridayThe13ths(1986));      // 1
console.log(fridayThe13ths(2015));      // 3
console.log(fridayThe13ths(2017));      // 2

File: launchschool_exercises/109_small_problems/medium2/unlucky_days.rb
require 'date'

def friday_13th(year)
  fridays = 0
  1.upto(12) do |month|
    fridays += 1 if Date.new(year, month, 13).friday?
  end
  fridays
end

p friday_13th(2015) == 3
p friday_13th(1986) == 1
p friday_13th(2019) == 2
